olp-cpp-sdk-dataservice-read/src/repositories/NamedMutex.h

```diff
@@ -64,6 +87,21 @@ class NamedMutex final {
   bool try_lock();
   void unlock();
 
+  /**
+   * @brief Saves an error to share it among threads.
+   *
+   * @param error An error to be stored.
+   */
+  void set_error(const client::ApiError& error);
+
+  /**
+   * @brief Gets the stored error for this mutex.
+   *
+   * @return The stored `ApiError` instance  or `boost::none` if no error has
+   * been stored for this mutex.
+   */
+  boost::optional<client::ApiError> get_error();
+
```

Comment:
Why are these methods lower camel case? The above method lock, try_lock, and unlock are lower snake case for the unique_lock.
Please add a comment above that these methods are to simulate mutex API and make the rest upper camel case.